node_modules/oae-authentication/lib/util.js

```diff
@@ -227,6 +230,40 @@ var validateRedirectUrl = module.exports.validateRedirectUrl = function(redirect
 };
 
 /**
+ * Get or create the user with the specified details as a user authenticates
+ * from an external authentication strategy. Since we trust emails from external
+ * authentication strategies, this method takes care of handling whether or not
+ * the specified email should be verified, as well as determines if there is any
+ * email invitation information that should be taken into consideration
+ *
+ * Aside from the `req` parameter, this function is simply a wrapper to
+ * `AuthenticationAPI.getOrCreateUser`
+ *
+ * @param  {Request}    req     The express request that holds the authentication info
+ * @see AuthenticationAPI.getOrCreateUser
+ */
+var handleExternalGetOrCreateUser = module.exports.handleExternalGetOrCreateUser = function(req, authProvider, externalId, providerProperties, displayName, opts, callback) {
+    if (!opts.email) {
+        // If no email was provided by the external authentication provider,
+        // we should provide the invitation token to indicate that the user
+        // should be created with their email address pre-validated to that
+        // associated to the invitation token
+        var invitationInfo = _getRequestInvitationInfo(req);
+        if (invitationInfo.invitationToken) {
+            opts.invitationToken = invitationInfo.invitationToken;
```

Comment:
Can you add a test case that tests this logic? There's `AuthenticationTestUtil.assertGoogleLoginSucceeds` that should help with this